yt_dlp/extractor/zattoo.py

```diff
@@ -243,7 +294,67 @@ def _make_valid_url(tmpl, host):
 
 
 class ZattooIE(ZattooBaseIE):
-    _VALID_URL_TEMPLATE = r'https?://(?:www\.)?%s/watch/(?P<channel>[^/]+?)/(?P<id>[0-9]+)[^/]+(?:/(?P<recid>[0-9]+))?'
+    _VALID_URL_TEMPLATE = r'''(?x)
+                            https?://(?:www\.)?%s/
+                            (?:
+                                (?:
+                                    vod/movies/([A-Za-z0-9]+)|
+                                    (?:program|watch)/([^/]+)/(\d+)|
+                                    live/([^/]+)
+                                )|
+                                (?:
+                                    .+\?recording=([0-9]+)|
+                                    .+\?movie_id=([A-Za-z0-9]+)|
+                                    .+\?channel=([^&]+)(?:
+                                        &program=(\d+)
+                                    )?
+                                )
+                            )'''
+
+    _VALID_URL = _make_valid_url(_VALID_URL_TEMPLATE, ZattooBaseIE._HOST)
```

Comment:
```suggestion
    _VALID_URL = r'''(?x)
        https?://(?:www\.)?zattoo.com/(?:
            (?:
                vod/movies/([A-Za-z0-9]+)|
                (?:program|watch)/([^/]+)/(\d+)|
                live/([^/]+)
            )|
            [^?#]+\?(?:
                recording=([0-9]+)|
                movie_id=([A-Za-z0-9]+)|
                channel=([^&]+)(?:&program=(\d+))?
            )
        )'''
```
Also:
1. Use named groups
2. Simplify regex to only include what is needed to perform a match. Rest of the data can be obtained using `parse_qs`

Reply:
this is still pending

Reply:
Yes, I have to understand this better. I'll have a look tomorrow.

Reply:
Here is my suggestion to simplify the regex: [a0126f0](https://github.com/yt-dlp/yt-dlp/pull/2288/commits/a0126f0152adfa7b32995c5ce73fd33921f234aa)